itou/www/signup/views.py

```diff
@@ -221,18 +221,7 @@ def setup(self, request, *args, **kwargs):
         super().setup(request, *args, **kwargs)
 
     def dispatch(self, request, *args, siae_id, **kwargs):
-        # TODO(FF): After a week on production (magic link validity), remove
-        # fallback logic to decode the siae_id and specify the parameter as
-        # siae_id:int in the urls.
-        try:
-            siae_id = int(siae_id)
-        except ValueError:
-            siae_id = urlsafe_base64_decode(siae_id)
-            try:
-                siae_id = int(siae_id)
-            except ValueError:
-                siae_id = None
-        self.siae = None if siae_id is None else Siae.objects.active().filter(pk=siae_id).get()
+        self.siae = Siae.objects.active().get(pk=siae_id)
         if self.siae is None or not siae_signup_token_generator.check_token(siae=self.siae, token=self.token):
```

Comment:
`self.siae` ne peut plus être `None` :)
Ou alors il faut faire un `.first()`.

Reply:
Haha, mon push-force a croisé ta revue ;)

Reply:
Mais il reste la 500 sur un mauvais SIAE ID.

Reply:
Oui, sachant qu'on peut supprimer une SIAE lors d'un merge, ça serait bien de le gérer plus proprement.
Le `.first()` avec le `.is None` ferait bien le travail a priori.

Reply:
Pas tout à fait, mais c’est le moment d’introduire `get_or_none`, ça fait un moment qu’il me titille.

Reply:
Après en vrai, un doublon sur `pk=siae_id` est heureusement impossible.